### PR TITLE
Worskops 3: Single Post Page

### PR DESCRIPTION
**Pull request by Bopiotr**

### CODE REVIEW

File: modules/posts/types.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 export interface Post {
   id: string;
   createdAt: string;
-  authorId: string;
+  authorNick: string;
```
Comment (dgrzedzielski) [inline modules/posts/types.ts]: We should have author username and avatar

File: modules/posts/types.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
   title: string;
   description: string;
   imageUrl: string;
+  commentsCount: number;
+  likesCount: number;
```
Comment (dgrzedzielski) [inline modules/posts/types.ts]: Remember that it could be also dislike

File: modules/posts/componets/single-post.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  display: flex;
+  max-width: 600px;
+  flex-direction: column;
+  padding: 16px;
+  border: var(--border-std);
+  border-radius: 12px;
+  background-color: var(--card-bg);
+`;
+
+export const SinglePost = ({ post }: { post: Post }) => {
```
Comment (Isterius) [inline modules/posts/componets/single-post.tsx:25]: where story?

File: modules/posts/hooks/use-post.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { useQuery } from 'react-query';
+
+import { getPost } from '../api/http.service';
+
+export const usePost = (postId: string) => {
+  return useQuery(['postId', postId], () => getPost(postId), {
```
Comment (dgrzedzielski) [inline modules/posts/hooks/use-post.ts]: i would use something like `['posts', postId]`

File: pages/posts/[id].tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  justify-content: center;
+`;
+
+const PostPage = () => {
+  const router = useRouter();
+  const { id } = router.query;
+  const { status, data } = usePost(id as string);
+
+  if (!data?.post) {
+    console.log(status);
```
Comment (dgrzedzielski) [inline pages/posts/[id].tsx]: console.log 👀

File: pages/posts/[id].tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    return <div>404</div>;
+  }
+
+  return (
+    <StyledContainer>
+      <Link href={`${+id - 1}`}>Prev</Link>
```
Comment (dgrzedzielski) [inline pages/posts/[id].tsx]: Our ids are string, we cannot do that

**Pull request merged**